Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -55,10 +68,45 @@ namespace MultiplayerSample
 
     void NetworkWeaponsComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
+        ;
     }
 
-    void NetworkWeaponsComponent::HandleSendConfirmHit([[maybe_unused]] AzNetworking::IConnection* invokingConnection, [[maybe_unused]] const WeaponIndex& WeaponIndex, [[maybe_unused]] const HitEvent& HitEvent)
+    void NetworkWeaponsComponent::HandleSendConfirmHit([[maybe_unused]] AzNetworking::IConnection* invokingConnection, const WeaponIndex& weaponIndex, const HitEvent& hitEvent)
     {
+        if (GetWeapon(weaponIndex) == nullptr)
+        {
+            AZLOG_ERROR("Got confirmed hit for null weapon index");
+            return;
+        }
+
+        WeaponHitInfo weaponHitInfo(*GetWeapon(weaponIndex), hitEvent);
+        OnWeaponConfirmHit(weaponHitInfo);
+    }
+
+    void NetworkWeaponsComponent::ActivateWeaponWithParams(WeaponIndex weaponIndex, WeaponState& weaponState, const FireParams& fireParams, bool validateActivations)
+    {
+        const uint32_t weaponIndexInt = aznumeric_cast<uint32_t>(weaponIndex);
+
+        // Temp hack for weapon firing due to late ebus binding in 1.14
+        if (m_fireBoneJointIds[weaponIndexInt] == InvalidBoneId)
+        {
+            const char* fireBoneName = GetFireBoneNames(weaponIndexInt).c_str();
+            m_fireBoneJointIds[weaponIndexInt] = GetNetworkAnimationComponent()->GetBoneIdByName(fireBoneName);
+        }
+
+        AZ::Transform fireBoneTransform;
+        if (!GetNetworkAnimationComponent()->GetJointTransformById(m_fireBoneJointIds[weaponIndexInt], fireBoneTransform))
+        {
+            AZLOG_WARN("Failed to get transform for fire bone %s, joint Id %u", GetFireBoneNames(weaponIndexInt).c_str(), m_fireBoneJointIds[weaponIndexInt]);
```

Comment:
Is the default just identity here? Is there a more meaningful alternative value here? Maybe just object transform?

Reply:
It essentially means the data is bad, but it's not unrecoverable. The simulation will behave incorrectly but it will not crash.